### ISSUE TITLE
Reproducibility of SDF

### ISSUE THREAD

**Issue by ZhengdiYu** [vsitzmann/siren#47]
Hey,

Thank you for your work.

![image](https://user-images.githubusercontent.com/63605407/124547691-6cc57180-de24-11eb-8265-6f17fdf7e3f2.png)

According to the paper, the signed distance function constraint as well as the off-surface penalization should be multiplied by the same λ. However, the λ of 'inter_constraint(off-surface)' and 'sdf_constraint' is totally different in the code. 

1. Shouldn't 'inter' be multiplied by 3e3? Which one should I follow?
2. Do you use a different hidden_features for thai and room?
3. Does 50000 iteration means 50000 steps or epochs?
4. The loss value is very unstable sometimes. (see below)

I tried using the original setting of your code to train the network on room and thai with inter_loss * 1e2 / inter_loss * 3e3.
A. room1e2, B. room3e3, C. thai1e2, D. thai3e3. 

**A, room1e2:** 
![image](https://user-images.githubusercontent.com/63605407/124548278-684d8880-de25-11eb-836b-e5d69710d5ef.png)

**B, room3e3:**
![image](https://user-images.githubusercontent.com/63605407/124549581-42c17e80-de27-11eb-997b-4daafa205f2f.png)

**C thai1e2:**
![image](https://user-images.githubusercontent.com/63605407/124548701-faee2780-de25-11eb-9761-71330075c6e7.png)

**D thai3e3:**
![image](https://user-images.githubusercontent.com/63605407/124549346-f24a2100-de26-11eb-84ad-fc8ad39e9c36.png)

As you can see, none of these has a perfect convergency. I use hidden_feautre=256 for all 4 experiments. So if the 'units' in your paper means 'hidden_feautres' in code, then **C thai1e2** should be exactly the same as you described. But as you can see, the inter_loss did not converge at all.

Best,
Zhengdi

**Comment by peihaowang** [comment]
Hi Zhengdi,

Did you solve your problem? I have encountered the reproducibility problem as well. Have you ever visualized the SDF or the mesh? Besides non-convergence, I also failed to reproduce them (the exported ply file basically contains no vertices!)

I would appreciate any comments from authors to help reproduce the results.

BTW, if I want to regress SDF on ShapeNet dataset, is there anything I need to note?

Thanks,
Peihao

**Comment by ZhengdiYu** [comment]
> Hi Zhengdi,
> 
> Did you solve your problem? I have encountered the reproducibility problem as well. Have you ever visualized the SDF or the mesh? Besides non-convergence, I also failed to reproduce them (the exported ply file basically contains no vertices!)
> 
> I would appreciate any comments from authors to help reproduce the results.
> 
> BTW, if I want to regress SDF on ShapeNet dataset, is there anything I need to note?
> 
> Thanks, Peihao

Hi,

Yes, I have visualize the mesh. It seems generally great even some of the loss didn't converge at all. I have no idea why would this happen. It's been a long time though, I do remember that there are some discrepancies between paper and code. And I just did both of them. I remember that we need to follow the code, not the paper. The first picture was following the code. Rest of them were following the setting mentioned in the paper:

![8dc49c092452a875eb58bf9aa2461bf](https://user-images.githubusercontent.com/63605407/148610472-8ae540d9-a150-4072-9451-f48040bdbba2.png)
![3c9f5c2b67c3e91e5d95016064db3ca](https://user-images.githubusercontent.com/63605407/148610636-795c1b59-8226-4936-81f7-f4c97eea1af9.png)
![3c9f5c2b67c3e91e5d95016064db3ca](https://user-images.githubusercontent.com/63605407/148610661-4154470a-e762-41ab-a37a-a59596effe0c.png)

**Comment by peihaowang** [comment]
Thanks for your prompt reply. I just found my problem was due to my installed `scikit-image` package does not have `marching_cubes_lewiner` method. I think in latest version `scikit-image`, this function was rename to `marching_cubes`.

And I also found if the point cloud is very sparse (e.g., objects from shapenet only have ~2k vertices), the regressed results will have lots of holes.

**Comment by LiuYuchao666** [comment]
> > Hi Zhengdi,
> > Did you solve your problem? I have encountered the reproducibility problem as well. Have you ever visualized the SDF or the mesh? Besides non-convergence, I also failed to reproduce them (the exported ply file basically contains no vertices!)
> > I would appreciate any comments from authors to help reproduce the results.
> > BTW, if I want to regress SDF on ShapeNet dataset, is there anything I need to note?
> > Thanks, Peihao
> 
> Hi,
> 
> Yes, I have visualize the mesh. It seems generally great even some of the loss didn't converge at all. I have no idea why would this happen. It's been a long time though, I do remember that there are some discrepancies between paper and code. And I just did both of them. I remember that we need to follow the code, not the paper. The first picture was following the code. Rest of them were following the setting mentioned in the paper:
> 
> ![8dc49c092452a875eb58bf9aa2461bf](https://user-images.githubusercontent.com/63605407/148610472-8ae540d9-a150-4072-9451-f48040bdbba2.png) ![3c9f5c2b67c3e91e5d95016064db3ca](https://user-images.githubusercontent.com/63605407/148610636-795c1b59-8226-4936-81f7-f4c97eea1af9.png) ![3c9f5c2b67c3e91e5d95016064db3ca](https://user-images.githubusercontent.com/63605407/148610661-4154470a-e762-41ab-a37a-a59596effe0c.png)

Hi Zhengdi,
I came across your reply. The Thai statue model you reproduced looks really great!

I'm currently trying to reproduce this work as well and have encountered some issues. I was wondering if you could share some insights:
1. About how long did it take you to reproduce that statue model back then? What hardware were you using?Using an RTX 8000 GPU, my training is estimated to take about 90 hours, while the paper mentions it should take only around 6 hours.
2. After 50k iterations, approximately what value did the loss function converge to?I'm currently at 10k iterations and the loss is still around 5, which seems quite slow to converge.

If you remember any details from your experience, I'd really appreciate it if you could share! :)
